test/CountedPtrBenchmarks.h

```diff
@@ -0,0 +1,282 @@
+                << " times without reference counting took: " << RawPtrCopyTime.count() << " Microseconds."
+                << "\n";
+
+        /// Internal Reference Counter
+        Begin = ClockType::now();
+        for( Whole Count = 0 ; Count < TestCount ; Count++ )
+        {
+            CountedPtr<FooInternal> PtrI( new FooInternal(&destructFlag, 7) );
+            CountedPtr<FooInternal> PtrI2( PtrI );
+            OutputI = PtrI2->Value;
+        }
+        End = ClockType::now();
+        DurationType CountPtrCopyInternalTime = End - Begin;
+        TestLog << OutputI << " - Creating, Dereferencing and Copying a CountedPtr " << TestCount
+                << " times with internal counting took: " << CountPtrCopyInternalTime.count() << " Microseconds."
+                << "\n";
+
+        /// External Reference Counter
+        Begin = ClockType::now();
+        for( Whole Count = 0 ; Count < TestCount ; Count++ )
+        {
+            CountedPtr<FooExternal> PtrE( new FooExternal(&destructFlag, 8) );
+            CountedPtr<FooExternal> PtrE2(PtrE);
+            OutputE = PtrE2->Value;
+        }
+        End = ClockType::now();
+        DurationType CountPtrCopyExternalTime = End - Begin;
+        TestLog << OutputE << " - Creating, Dereferencing and Copying a CountedPtr " << TestCount
+                << " times with external counting took: " << CountPtrCopyExternalTime.count() << " Microseconds."
+                << "\n";
+
+
+        /// shared_ptr
+        Begin = ClockType::now();
+        for( Whole Count = 0 ; Count < TestCount ; Count++ )
+        {
+            std::shared_ptr<FooExternal> PtrS( new FooExternal(&destructFlag, 9) );
+            std::shared_ptr<FooExternal> PtrS2(PtrS);
+            OutputS = PtrS2->Value;
+        }
+        End = ClockType::now();
+        DurationType SharedPtrCopyTime = End - Begin;
+        TestLog << OutputS << " - Creating, Dereferencing and Copying a shared_ptr " << TestCount
+                << " times with external counting took: " << SharedPtrCopyTime.count() << " Microseconds."
+                << "\n";
+
+        /// make_shared
+        Begin = ClockType::now();
+        for( Whole Count = 0; Count < TestCount ; Count++)
+        {
+            std::shared_ptr<FooExternal> PtrS = std::make_shared<FooExternal>(&destructFlag, 10);
+            std::shared_ptr<FooExternal> PtrS2(PtrS);
+            OutputS = PtrS2->Value;
+        }
+        End = ClockType::now();
+        DurationType MakeSharedCopyTime = End - Begin;
+        TestLog << OutputS << " - Creating, Dereferencing and Copying a shared_ptr from make_shared "
+                << TestCount << " times with external counting took: " << MakeSharedCopyTime.count() << " Microseconds."
+                << "\n\n";
+
+        /////////////////////////////////////
+        // Comparison of Benchmark Results
+
+        TestLog << "Checking that raw pointers are fastest for sanity: "
```

Comment:
Consider sending std::boolalpha to turn "1" and "0" in the output into "true" and "false".